h/services/bulk_api/lms_stats.py

```diff
@@ -71,85 +77,102 @@ def _annotation_query(self, groups: list[str]):
                 Group.authority_provided_id.in_(groups),
             )
         )
-
-    def get_counts_by_user(
-        self, groups: list[str], assignment_id: str
-    ) -> list[CountsByUser]:
-        """
-        Get basic stats per user for an LMS assignment.
-
-        :param groups: List of "authority_provided_id" to filter groups by.
-        :param assignment_id: ID of the assignment we are generating the stats for.
-        """
-        annos_query = (
-            self._annotation_query(groups)
-            .where(
+        if assignment_id:
+            query = query.where(
                 AnnotationMetadata.data["lms"]["assignment"]["resource_link_id"].astext
                 == assignment_id,
             )
-            .cte("annotations")
-        )
 
-        query = (
-            select(
-                User.display_name,
-                # Unfortunately all the magic around User.userid doesn't work in this context
-                func.concat("acct:", User.username, "@", User.authority).label(
-                    "userid"
-                ),
-                func.count(annos_query.c.id)
-                .filter(annos_query.c.type == "annotation")
-                .label("annotations"),
-                func.count(annos_query.c.id)
-                .filter(annos_query.c.type == "reply")
-                .label("replies"),
-                func.max(annos_query.c.created).label("last_activity"),
+        if h_userids:
```

Comment:
New filter by user.

This becomes useful right now to exclude teachers from course counts but we can use the same parameter to implement the filters that we are planning to tackle next.